lib/pbench/test/unit/server/test_requests.py

```diff
@@ -88,109 +88,139 @@ def test_json_object(client, caplog, server_config):
 
 
 class TestUpload:
+    @pytest.fixture
+    def setup_ctrl(self):
+        self.controller = socket.gethostname()
+        yield
+        self.controller = None
+
     @staticmethod
-    def test_missing_authorization_header(client, caplog, server_config):
-        response = client.put(
-            f"{server_config.rest_uri}/upload/ctrl/{socket.gethostname()}",
-            headers={"filename": "f.tar.xz"},
-        )
+    def gen_uri(server_config, filename="f.tar.xz"):
+        return f"{server_config.rest_uri}/upload/{filename}"
+
+    def gen_headers(self, auth_token, md5=None):
+        headers = {
+            "Authorization": "Bearer " + auth_token,
+            "controller": self.controller,
+        }
+        if md5:
+            headers["Content-MD5"] = md5
+        return headers
+
+    def test_missing_authorization_header(self, client, caplog, server_config):
+        response = client.put(self.gen_uri(server_config))
         assert response.status_code == 401
         for record in caplog.records:
-            assert record.levelname not in ("ERROR", "CRITICAL")
+            assert record.levelname not in ("DEBUG", "ERROR", "CRITICAL")
+        assert caplog.records[-1].levelname == "WARNING"
 
-    @staticmethod
```

Comment:
Just out of curiosity, why remove the `@staticmethod`?  (The only reference that I see to `self` here is the invocation of `gen_uri()` which is also a static method, so you can invoke it as `TestUpload.gen_uri()` without having to make this an instance method.)

Reply:
I got into a battle with `pytest` on how to use instance attributes during tests.  I nearly lost, but the result of that skirmish was this change.  I think I could change it back, but am really not up for dealing with that again.